pkg/segment/writer/segwriter.go

```diff
@@ -1174,19 +1187,159 @@ func DeletePQSData() error {
 	return pqsmeta.DeletePQMetaDir()
 }
 
-func CreateDci(sIdx uint32, wlen uint16, recBsIdx uint16) *DwordCbufIdxs {
+func (ss *SegStore) writeToBloom(encType []byte, buf []byte, cname string,
+	cw *ColWip) error {
+
+	// no bloom for timestamp column
+	if encType[0] == TIMESTAMP_TOPDIFF_VARENC[0] {
+		return nil
+	}
+
+	bi, ok := ss.wipBlock.columnBlooms[cname]
+	if !ok {
+		// for non-strings columns, BI is not iniliazed. Maybe there should be an
+		// explicit way of saying what columnType is this so that we don't "overload" the BI var
+		return nil
+	}
+
+	switch encType[0] {
+	case ZSTD_COMLUNAR_BLOCK[0]:
+		return cw.writeNonDeBloom(buf, bi, ss.wipBlock.blockSummary.RecCount, cname)
+	case ZSTD_DICTIONARY_BLOCK[0]:
+		return cw.writeDeBloom(buf, bi)
+	default:
+		log.Errorf("writeToBloom got an unknown encoding type: %+v", encType)
+		return fmt.Errorf("got an unknown encoding type: %+v", encType)
+	}
+}
 
-	dci := &DwordCbufIdxs{
-		sIdx:     sIdx,
-		wlen:     wlen,
-		recBsIdx: recBsIdx,
+func (cw *ColWip) writeDeBloom(buf []byte, bi *BloomIndex) error {
+	// todo a better way to size the bloom might be to count the num of space and
+	// then add to the cw.deData.deCount, that should be the optimal size
+	// we add twice to avoid undersizing for above reason.
+	bi.Bf = bloom.NewWithEstimates(uint(cw.deData.deCount), BLOOM_COLL_PROBABILITY)
```

Comment:
I'm not seeing where we "add twice." Where is this happening?

Reply:
good catch, I was doing it for NonDeBloom, but forgot to do it for DeBloom. Fixed it